source/common/tcp_proxy/upstream.cc

```diff
@@ -77,12 +77,57 @@ TcpUpstream::onDownstreamEvent(Network::ConnectionEvent event) {
 
 HttpUpstream::HttpUpstream(Tcp::ConnectionPool::UpstreamCallbacks& callbacks,
                            const TunnelingConfigHelper& config,
-                           StreamInfo::StreamInfo& downstream_info)
+                           StreamInfo::StreamInfo& downstream_info, Http::CodecType type)
     : config_(config), downstream_info_(downstream_info), response_decoder_(*this),
-      upstream_callbacks_(callbacks) {}
+      upstream_callbacks_(callbacks), type_(type) {}
 
 HttpUpstream::~HttpUpstream() { resetEncoder(Network::ConnectionEvent::LocalClose); }
 
+bool HttpUpstream::isValidResponse(const Http::ResponseHeaderMap& headers) {
+  if (type_ == Http::CodecType::HTTP2) {
+    if (Http::Utility::getResponseStatus(headers) != 200) {
+      return false;
+    }
+    return true;
+  }
```

Comment:
I think beforehand this was running for type_ both H2 and H3? from your comments in https://github.com/envoyproxy/envoy/pull/30725/files/433a95dc85e017cfe35dfd13f7340cf0e8960f74#r1386014784

Reply:
for H3 it does not get invoked. 
Still to keep the logic same as beforehand, will change the logic